rust/worker/src/execution/operators/get_vectors_operator.rs

```diff
@@ -112,8 +112,10 @@ impl Operator<GetVectorsOperatorInput, GetVectorsOperatorOutput> for GetVectorsO
         let logs = input.log_records.clone();
         let mut remaining_search_user_ids: HashSet<String> =
             HashSet::from_iter(input.search_user_ids.iter().cloned());
+        let mut original_search_user_ids: HashSet<String> =
+            HashSet::from_iter(input.search_user_ids.iter().cloned());
         for (log_record, _) in logs.iter() {
-            if remaining_search_user_ids.contains(&log_record.record.id) {
+            if original_search_user_ids.contains(&log_record.record.id) {
```

Comment:
Why is this needed? Isn't there only one entry in the log after materialization?

Reply:
I think you made an explicit decision to not materialize here (although I think we've pretty much implemented 40%-50% of materialization here)

Reply:
Oh woops - confused myself :) thanks.